genimage.c

```diff
@@ -742,6 +746,7 @@ int main(int argc, char *argv[])
 		image->name = cfg_getstr(imagesec, "name");
 		image->size = cfg_getint_suffix_percent(imagesec, "size",
 				&image->size_is_percent);
+		image->srcpath = cfg_getstr(imagesec, "srcpath");
 		image->mountpoint = cfg_getstr(imagesec, "mountpoint");
```

Comment:
Add the check here: If `srcpath` is not `NULL` then `mountpoint` must be `NULL` or `"/"`. Otherwise print an error and `goto cleanup`.

Or maybe below, when `mountpoint` is manipulated.

Reply:
i believe i cannot use `goto cleanup;` in context of `mountpath()` (or anywhere outside of `main()`)...

Reply:
I mean, add the check here in `main()` or a few lines below, also in `main()`:
```
if (image->mountpoint && *image->mountpoint == '/')
	image->mountpoint++;
if (image->srcpath && image->mountpoint && (strlen(image->mountpoint) > 0)) {
	image_error(...);
	goto cleanup;
}

Reply:
Fixed